resources/azure/environmentDefaults.go

```diff
@@ -43,8 +49,12 @@ func sandboxDefault() environmentDefault {
 			defaultVNet:            "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/virtualNetworks/default-vnet",
 			defaultSubnet:          "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/virtualNetworks/default-vnet/subnets/default-subnet",
 			defaultSecurityGroup:   "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/networkSecurityGroups/default",
-			defaultInstanceType:    "Standard_B4ms",
-			defaultARMInstanceType: "Standard_D4ps_v5",
+			defaultInstanceType:    "Standard_D4ps_v5",
+			defaultARMInstanceType: "Standard_D4ps_v5", // No azure arm instance supports nested virtualization
+			aks: ddInfraAks{
+				defaultKataInstanceType: "Standard_D4s_v3", // Allows nested virtualization for kata runtimes
```

Comment:
I think you can remove that as well and set a default that supports nested virtualization (done above)

Reply:
Done in the last commit